### ISSUE TITLE
Allow to disable backups on battery power

### ISSUE THREAD

**Issue by sidevesh** [zhangyuannie/butter#19]
This is something that Mac OS does for TIme Machine backups by default and allows this to be toggled off if needed.

**Comment by zhangyuannie** [comment]
Do you have an idea how much it will save in term of power? AFAIK unlike other real backup tools, btrfs snapshot is extremely cheap so I don't think it will make any meaningful difference.

**Comment by sidevesh** [comment]
You are right, all the issues I have opened are basically to match Butter up to Time Machine on Mac OS and it has a setting to not backup on battery power,
but I guess that's because Time Machine backups are meant to backup to External NAS or USB storage,
it just so happens that it does that by first taking a snapshot like BTRFS snapshots and it also keeps those snapshots as long as they remain within space threshold so that limited / local File history is also available even if backup drive is not reachable.
Now since Butter does not have backup to external storage yet, I guess we don't really need this until that gets implemented. We can keep this open until then or close it, up to you.

I had seen your comment on reddit mentioning that you plan to add support for moving snapshots to external storage,
what approach were you considering for that ?
I am planning of setting that up to and there doesn't seem to be any way to do that as seamlessly as Time Machine on Mac OS but I have a few approaches in mind.

**Comment by sidevesh** [comment]
@zhangyuannie I have added the battery check in a PR,
addressing your earlier comment, while btrfs snapshots are extremely cheap,
the battery check right now would be more useful if pre or post snapshot hooks are being used and if they are heavy on battery,
my usecase, which I think would be very standard, would be to move the backups post snapshot to an external NAS and doing that on battery might be intensive.
